shared/middleware/helpers/bucket_helper.rb

```diff
@@ -338,19 +338,23 @@ def hard_delete_channel_content(encrypted_channel_id)
     result.deleted.count
   end
 
-  def list_versions(encrypted_channel_id, filename)
+  def list_versions(encrypted_channel_id, filename, with_comments: false)
     owner_id, storage_app_id = storage_decrypt_channel_id(encrypted_channel_id)
     key = s3_path owner_id, storage_app_id, filename
 
     track_list_operation 'BucketHelper.list_versions'
     s3.list_object_versions(bucket: @bucket, prefix: key).
       versions.
       map do |version|
+        comment = with_comments ?
+          DASHBOARD_DB[:project_versions].select(:comment).where(storage_app_id: storage_app_id, object_version_id: version.version_id).first&.fetch(:comment) :
```

Comment:
.first guaranteed to work even when no results (ie, no safe navigation necessary)?

Reply:
And for my understanding, this is sequel? Would be nice to split this onto multiple lines if possible, but maybe causes issues in a ternary?

Reply:
> .first guaranteed to work even when no results (ie, no safe navigation necessary)?

Yes, I believe `where` always returns an array so `first` will work, but might itself return `nil`. Below is an excerpt from calling this in my browser. Some versions have a comment, some don't.
```{"versionId":"xoIslfy9nAZuXxpn.eQrcC8.g.O387KN","lastModified":"2021-08-12T16:33:11.000Z","comment":"hi","isLatest":false},{"versionId":"es3D.s0lw3r3oH5FdM.fwQ88cLzUGos5","lastModified":"2021-08-12T16:30:42.000Z","isLatest":false},{"versionId":"TbMd14CRSoJ6iw8hIbWt7HmTF.AKJOds","lastModified":"2021-08-12T16:25:49.000Z","isLatest":false},```

Reply:
I can break up the line (sorry got distracted)